execution/gethexec/caff_node.go

```diff
@@ -83,35 +82,29 @@ func (n *CaffNode) createBlock() (returnValue bool) {
 
 	lastBlockHeader := n.executionEngine.bc.CurrentBlock()
 
-	currentMsgPos, err := n.executionEngine.BlockNumberToMessageIndex(lastBlockHeader.Number.Uint64())
-	if err != nil {
-		log.Error("failed to convert block number to message index")
-		return false
-	}
-	currentPos := uint64(currentMsgPos)
+	refreshed, err := n.espressoStreamer.Refresh(ctx, func() (uint64, uint64, error) {
```

Comment:
I think we're not supposed to be calling refresh here. We know that the caff node always want the next message position from the streamer every time we call Next() unless we have seen produce block fail. At this point, we no longer have the message position we need in the streamers buffer, and we should refresh to get it back. Otherwise we want to continue normally by parsing the next message.

Reply:
Yes thats what the code is doing? Its only calling refresh when there is a need for refresh otherwise `refreshed` returns false. Its true only when there was a failure to write in the database.